snuba/snapshots/postgres_snapshot.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import jsonschema  # type: ignore
+import json
+import logging
+
+from contextlib import contextmanager
+from dataclasses import dataclass
+from os import path as os_path
+from typing import NewType, Generator, IO, Sequence
+
+from snuba.snapshots import SnapshotDescriptor, SnapshotId, TableConfig
+from snuba.snapshots import BulkLoadSource
+
+Xid = NewType("Xid", int)
+
+
+@dataclass(frozen=True)
+class PostgresSnapshotDescriptor(SnapshotDescriptor):
+    """
+    Provides the metadata for the loaded snapshot.
+    """
+    id: SnapshotId
+    xmin: Xid
+    xmax: Xid
+    xip_list: Sequence[Xid]
+    tables: Sequence[TableConfig]
```

Comment:
Are `id` and `tables` not inherited from the parent class? I haven't played with `dataclass` inheritance.

Reply:
yes, this is a mistake